pkg/controller/logstorage/esgateway.go

```diff
@@ -47,35 +49,32 @@ func (r *ReconcileLogStorage) createEsGateway(
 	svcDNSNames = append(svcDNSNames, dns.GetServiceDNSNames(esgateway.ServiceName, render.ElasticsearchNamespace, r.clusterDomain)...)
 	gatewayKeyPair, err := certificateManager.GetOrCreateKeyPair(r.client, render.TigeraElasticsearchGatewaySecret, common.OperatorNamespace(), svcDNSNames)
 	if err != nil {
-		log.Error(err, "Error creating TLS certificate")
-		r.status.SetDegraded("Error creating TLS certificate", err.Error())
+		status.SetDegraded(r.status, operatorv1.ResourceCreateError, "Error creating TLS certificate", err, reqLogger)
 		return reconcile.Result{}, false, err
 	}
 	kibanaCertificate, err := certificateManager.GetCertificate(r.client, render.TigeraKibanaCertSecret, common.OperatorNamespace())
 	if err != nil {
-		reqLogger.Error(err, "failed to get Kibana tls certificate secret")
-		r.status.SetDegraded("Failed to get Kibana tls certificate secret", err.Error())
+		status.SetDegraded(r.status, operatorv1.ResourceReadError, "Failed to get Kibana tls certificate secret", err, reqLogger)
 		return reconcile.Result{}, false, err
 	} else if kibanaCertificate == nil {
 		reqLogger.Info("Waiting for internal Kibana tls certificate secret to be available")
-		r.status.SetDegraded("Waiting for internal Kibana tls certificate secret to be available", "")
+		r.status.SetDegraded(string(operatorv1.ResourceNotReady), "Waiting for internal Kibana tls certificate secret to be available")
 		return reconcile.Result{}, false, nil
 	}
 	esInternalCertificate, err := certificateManager.GetCertificate(r.client, render.TigeraElasticsearchInternalCertSecret, common.OperatorNamespace())
 	if err != nil {
-		reqLogger.Error(err, "failed to get Elasticsearch tls certificate secret")
-		r.status.SetDegraded("Failed to get Elasticsearch tls certificate secret", err.Error())
+		status.SetDegraded(r.status, operatorv1.ResourceReadError, "Failed to get Elasticsearch tls certificate secret", err, reqLogger)
 		return reconcile.Result{}, false, err
 	} else if esInternalCertificate == nil {
 		reqLogger.Info("Waiting for internal Elasticsearch tls certificate secret to be available")
-		r.status.SetDegraded("Waiting for internal Elasticsearch tls certificate secret to be available", "")
+		r.status.SetDegraded(string(operatorv1.ResourceNotReady), "Waiting for internal Elasticsearch tls certificate secret to be available")
```

Comment:
You can use status.SetDegraded(). Please check for all controllers to see if r.status.SetDegraded() is still used.

Reply:
done